firewalldb/actions_kvdb.go

```diff
@@ -54,9 +55,32 @@ var (
 )
 
 // AddAction serialises and adds an Action to the DB under the given sessionID.
-func (db *BoltDB) AddAction(_ context.Context,
+func (db *BoltDB) AddAction(ctx context.Context,
 	req *AddActionReq) (ActionLocator, error) {
 
+	// If the new action links to a session, the session must exist.
+	// For the bbolt impl of the store, this is our best effort attempt
+	// at ensuring each action links to a session. If the session is
+	// deleted later on, however, then the action will still exist.
+	var err error
+	req.SessionID.WhenSome(func(id session.ID) {
+		_, err = db.sessionIDIndex.GetSession(ctx, id)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// If the new action links to an account, the account must exist.
+	// For the bbolt impl of the store, this is our best effort attempt
+	// at ensuring each action links to an account. If the account is
+	// deleted later on, however, then the action will still exist.
+	req.AccountID.WhenSome(func(id accounts.AccountID) {
+		_, err = db.accountsDB.Account(ctx, id)
+	})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
nit:
Same here but for `accounts.ErrAccNotFound`.

Reply:
already handled - see unit tests that assert this